### ISSUE TITLE
Feature idea: Reload button

### ISSUE THREAD

**Issue by gorjious** [arrowtype/type-x#112]
I find myself constantly through the cycle of exporting from Glyphs then going through the Type-X menu to select the newly exported font and then wait to see the changes. 

1. Click Type-X icon
2. Select Font
3. Click upload icon
4. Select font file
5. Click open
6. Click somewhere to close Type-X menu

It is actually quite a lot of steps if you do it constantly. I've set up shortcuts that replicate this process but I wonder if it would be possible to include a `Reload` button that will essentially know which font is activated and reload that font. If this Reload button could be triggered by a shortcut then that would make it even faster to test/retest fonts.

**Comment by arrowtype** [comment]
Ohhh this is a really great idea. It would make this a much more seamless tool for a type design workflow.

The only other way to improve it might be if type-x could actually make it automatic, watching for the file to change, and hot-reloading when it does. But, I suspect that might be limited by browser/OS security. If so, a reload button w/ shortcuts would be the next best thing.

**Comment by gorjious** [comment]
Automatic would be terrific, especially if there is a 2nd monitor so that after export, all you have to do is look to the browser screen and see the change. However, there is something to reloading manually and watching the change occur.

Ideally, the [per-tab](https://github.com/arrowtype/type-x/issues/105) feature for A/B testing is a better gauge to see how something changes.

If automatic is possible then there would be a conflict with the per-tab A/B testing idea because how would you decide what is automatically updated. 

Maybe there could be a way to switch between automatic/manually modes since both can be used for different purposes.

**Comment by justinpenner** [comment]
Agreed, this would be my top feature request for Type-X. An option to auto-reload fonts would be useful, as would a manual refresh button.

I'd imagine a manual refresh button next to each font in the list, next to the expand/collapse toggle, so it can be done in 2 clicks. I'd guess that most users using the auto-reload setting would want it applied to all fonts, so a toggle for global auto-reload could be somewhere below the list of fonts. You could still have individual toggles for each font to auto-reload, but toggling the global setting would override them all.

**Comment by RoelN** [comment]
I'm pretty sure auto-reload isn't going to work, as extensions aren't allowed to watch the user's filesystem. This would mean the extension can never be made aware of changes to the font file. The extensions I found that _do_ offer some kind of "hot reloading" need a server to run in the background, which does the actual watching. This would be possible for Type-X too, but require a lot more work (especially to create versions for all OSes).

A manual refresh button might have a better chance at succeeding (not sure about permissions etc. for file uploads triggered from code), which would eliminate step 3 to 6 in @gorjious post.

**Comment by slobzheninov** [comment]
Hi! Came here with the same feature suggestion and found this thread, so wanted to add one more vote for the refresh button.
Thanks for the extremely useful plugin anyway! Would be happy to thank in some donation, why is there no button :)
